File: backend/parking-domain/src/main/java/com/chatchatabc/parking/domain/service/InvoiceService.java
package com.chatchatabc.parking.domain.service;

import com.chatchatabc.parking.domain.model.Invoice;

public interface InvoiceService {

    /**
     * Create invoice for vehicle
     *
     * @param parkingLotId the parking lot id
     * @param vehicleId    the vehicle id
     * @return the invoice
     */
    Invoice createInvoice(String parkingLotId, String vehicleId);

    /**
     * End invoice
     *
     * @param invoiceId    the invoice id
     * @param parkingLotId the parking lot id
     * @return the invoice
     */
    Invoice endInvoice(String invoiceId, String parkingLotId);

    /**
     * Pay invoice
     *
     * @param invoiceId    the invoice id
     * @param parkingLotId the parking lot id
     * @return the invoice
     */
    Invoice payInvoice(String invoiceId, String parkingLotId);
}

File: backend/parking-service/src/main/kotlin/com/chatchatabc/parking/domain/service/InvoiceService.kt
package com.chatchatabc.parking.domain.service

import com.chatchatabc.parking.domain.model.Invoice
import org.springframework.stereotype.Service

@Service
interface InvoiceService {

    /**
     * Create an invoice
     */
    fun createInvoice(parkingLotId: String, vehicleId: String): Invoice

    /**
     * End an invoice
     */
    fun endInvoice(parkingLotId: String, invoiceId: String): Invoice

    /**
     * Pay an invoice
     */
    fun payInvoice(parkingLotId: String, invoiceId: String): Invoice
}